django/db/models/expressions.py

```diff
@@ -1068,12 +1076,11 @@ def as_sql(self, compiler, connection, template=None, **extra_context):
             sql = 'NOT {}'.format(sql)
         return sql, params
 
-    def as_oracle(self, compiler, connection, template=None, **extra_context):
+    def select_format(self, compiler, sql, params):
         # Oracle doesn't allow EXISTS() in the SELECT list, so wrap it with a
-        # CASE WHEN expression. Change the template since the When expression
-        # requires a left hand side (column) to compare against.
-        sql, params = self.as_sql(compiler, connection, template, **extra_context)
-        sql = 'CASE WHEN {} THEN 1 ELSE 0 END'.format(sql)
+        # CASE WHEN expression.
+        if compiler.connection.vendor == 'oracle':
```

Comment:
I like this approach much better than #11641 !

At first I was a little concerned about checking the vendor attribute within a vendor neutral method, but it's easy to patch on classes you don't control and easy to modifying on ones that you do.

If it ends up being an issue in practise moving forward we can look at better ways to make it more generic.

Reply:
@jarshwah my main concern is that it's somewhat of an abuse of `select_format` from my understanding as it's meant to format only top level select expressions.

https://github.com/django/django/blob/af385cc5686c8be0364db6bed526700ab296fc65/django/db/models/sql/compiler.py#L411

That means that an `Exists` in a subquery (e.g. nested `OuterRef`) would fail on Oracle.

e.g.

```sql
SELECT *
FROM (
   SELECT EXISTS(...) FROM ...
)
```

I guess we could change the signature of `Expression.select_format` to pass a `subquery` or `nested` boolean kwarg instead and have the default implementation return the the SQL untouched when `nested=True`.

Reply:
Weird, I wonder why it's not applicable to select section of subqueries. More importantly, I wonder if it's not applied to subqueries because it breaks something, or if it was to minimise the number of places that required it.

What happens when a geom field is used within a subquery or an order by clause?

Reply:
> Weird, I wonder why it's not applicable to select section of subqueries. More importantly, I wonder if it's not applied to subqueries because it breaks something, or if it was to minimise the number of places that required it.

From my a little bit of investigation it seems to be done for optimizaton purposes.

> What happens when a geom field is used within a subquery or an order by clause?

I don't think it really matters. Judging by the failures this formatting is only used to allow values to be retrieved in a format Django expects; the database can deal with unformated values just fine in subqueries and ordering.